tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -0,0 +1,290 @@
+import stat
+import pandas as pd
+import numpy as np
+
+from tardis.plasma.properties.base import ProcessingPlasmaProperty
+
+__all__ = [
+    "NLTERateEquationSolver",
+]
+
+
+class NLTERateEquationSolver(ProcessingPlasmaProperty):
+    outputs = ("ion_number_density_nlte", "electron_densities_nlte")
+
+    def calculate(
+        self,
+        gamma,
+        alpha_sp,
+        alpha_stim,
+        coll_ion_coeff,
+        coll_recomb_coeff,
+        partition_function,
+        levels,
+        level_boltzmann_factor,
+        phi,
+        rate_matrix_index,
+        number_density,
+    ):
+
+        (
+            photo_ion_rate,
```

Comment:
`photo_ion_rate ` is also a rate coefficient. Also, the naming scheme seems a bit inconsistent. We have coeff, coefficient, and nothing. I would recommend to use coefficient or coeff for all of them.

Reply:
Maybe we should add something like "species" or "total" to the names to distinguish them from the level-resolved quantities?

Reply:
Done

Reply:
Renamed all the coefficients to `total_(name)_coefficients`